src/main/java/seedu/address/logic/commands/schedule/EditScheduleCommand.java

```diff
@@ -114,40 +133,43 @@ public boolean equals(Object other) {
         }
 
         // instanceof handles nulls
-        if (!(other instanceof RescheduleCommand)) {
+        if (!(other instanceof EditScheduleCommand)) {
             return false;
         }
 
         // state check
-        RescheduleCommand e = (RescheduleCommand) other;
-        return scheduleIndex.equals(e.scheduleIndex)
-                && editRescheduleDescriptor.equals(e.editRescheduleDescriptor);
+        EditScheduleCommand e = (EditScheduleCommand) other;
+        return clientIndex.equals(e.clientIndex)
+                && sessionIndex.equals(e.sessionIndex)
+                && editScheduleDescriptor.equals(e.editScheduleDescriptor);
     }
 
     /**
      * Stores the details to edit the Schedule with. Each non-empty field value will replace the
      * corresponding field value of the Schedule.
      */
-    public static class RescheduleDescriptor {
+    public static class EditScheduleDescriptor {
         private Index clientIndex;
         private Index sessionIndex;
```

Comment:
are these 2 fields used? In editClientDescriptor, the index of the entity that we want to edit is not included. Only the fields that might be changed due to the edit command are included

Reply:
But I think for editclientdescriptor, the index is to filter our the client, but for my case, both client and session index are required to filter out the schedule just like how your deleteSchedule is like, it filters out the relevant schedule before it can be deleted